### DISCUSSION TITLE
Package Installation Issues, and Certificate Error - Output of Doctor.py

### DISCUSSION THREAD

**Discussion by KBagal123** [biostars/biostar-handbook-issues#292] [Q&A]
Hi! I am very new to bioinformatics and needed some help regarding the software installation for the BioStar Handbook. After installation, and running the doctor.py script, the following output presents:

```
(bioinfo) Kavins-MacBook-Air:~ kbagal$ ~/bin/doctor.py
# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa           ... OK
# datamash      ... OK
# fastqc -h     ... OK
# hisat2        ... OK
# seqret -h     ... OK
# featureCounts ... OK
# efetch        ... OK
# esearch       ... OK
# samtools      ... OK
# fastq-dump    ... OK
# bowtie2       ... OK
# bcftools      ... OK
# seqtk         ... OK
# seqkit        ... OK
# bio           --> ERROR
# fastq-dump -X 1 -Z SRR1553591 --> ERROR
#
# Found some problems
#
# Unable to run: bio, fastq-dump -X 1 -Z SRR1553591
#
# Check the Biostar Handbook for installation help.
#
#
# Warning: the ~/bin folder is not in your PATH!
```

(I also am not sure about the additional warning about the ~/bin folder not being in my PATH or how to fix it)

When running the `fastq-dump -X 1 -Z SRR1553591` command on its own, the following error presents:

```
(bioinfo) Kavins-MacBook-Air:~ kbagal$ fastq-dump -X 1 -Z SRR1553591
2023-10-19T08:52:15 fastq-dump.2.8.0 sys: connection failed while opening file within cryptographic module - mbedtls_ssl_handshake returned -9984 ( X509 - Certificate verification failed, e.g. CRL, CA or signature check failed )
2023-10-19T08:52:15 fastq-dump.2.8.0 sys: mbedtls_ssl_get_verify_result returned 0x8 (  !! The certificate is not correctly signed by the trusted CA  )
2023-10-19T08:52:15 fastq-dump.2.8.0 err: item not found while constructing within virtual database module - the path 'SRR1553591' cannot be opened as database or table
```

When attempting to install the bio package on its own the following runs, takes ~5 mins, and fails to download, presenting the following:

```
(bioinfo) Kavins-MacBook-Air:~ kbagal$ conda install bio
Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - bio

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://conda.anaconda.org/bioconda/osx-64
  - https://conda.anaconda.org/bioconda/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

However, I am not sure why the current channels do not work as both conda-forge and bioconda are installed, unless bio needs to be in a different environment from bioinfo to work. I've tried checking some of the other packages and checking the path, but the rest seemed to work. I also did not spot any other discussion that fit similar situations to mine, and the ones that seemed somewhat similar had fixes that didn't work as well. For example, I've tried downloading the previously mentioned NCBI native client in another announcement (`curl http://data.biostarhandbook.com/sratools.sh | bash`) to no avail (the same error above presents), and the similar comments either didn't work for me, or I didn't fully understand what fix they were implementing, and how to apply it to my situation. Sorry if this ends up being a repeated question due to my lack of understanding, but I hope that someone can illuminate some light on the issues that I have been having, or let me know what other information I can provide. Thanks in advance!

**Comment by ialbert** [comment]
there are two different issues at play, first the `bio` package needs to be installed with

    pip install bio

and not conda, 

the second error sounds like you have installed sra-tools with conda as well, that version does not work, you should uninstall the conda version of sra-tools and follow our install procedure

see the chapter on step by step install, specifically the sratools install on section 10 in 

https://www.biostarhandbook.com/step-by-step.html

**Comment by bunnyeatersdeluxe** [comment]
Keep in mind that everyone that follows the Windows installation guide gets this same error when following step-by-step in the book; it should be appropriately updated to avert issues such as this as I have seen many other people encounter the same exact issue.

**Reply by ialbert** [reply]
@bunnyeatersdeluxe I need a little more help here, 

I looked at the page and I was unable find a source of potential errors.

The original poster tried to use `conda install bio` but the step-by-step instructions show `pip install bio` as they should. There may be something else wrong on the page, but not the installation of `bio`. 

I have recently upgraded my home system to a brand new Windows 11 PC and the default, fully automated installation went through with no errors.

Installation errors are difficult to test and troubleshoot because once it works it is harder to undo an installation to evaluate it again.

**Reply by bunnyeatersdeluxe** [reply]
Thanks, the issue manifests itself when they follow the raw book - use the descriptions provided - and then they encounter the error OP posted - therefore they have to go to the website link you provided with the handier, and more up to date information and steps to avert such an issue.

I should clarify I was using Windows 10 and the issue is replicated when following the instructions per the handbook where doctor.py presents the exact error.

**Reply by ialbert** [reply]
the codes in the PDF versions of the book go out of date

unfortunately, the code needs to change on a regular basis to keep up - bioinformatics is a moving target

**Reply by bunnyeatersdeluxe** [reply]
Gotcha, just reporting to let you know and help keep the book up-to-date. It did say to report it after all!

**Reply by ialbert** [reply]
yes, thanks for the note ehre,

it is helpful to have it reported here as well, so that other people that might have the same problem find answers
